batch/batches/Delete/Engines/KDeletingCategoryUserSubscriberEngine.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * @package Scheduler
+ * @subpackage Delete
+ */
+class KDeletingCategoryUserSubscriberEngine extends KDeletingEngine
+{
+	const CATEGORY_SUBSCRIBE = 'CATEGORY_SUBSCRIBE';
+	
+	/* (non-PHPdoc)
+	 * @see KDeletingEngine::delete()
+	 */
+	protected function delete(KalturaFilter $filter)
+	{
+		$categoryId = $filter->categoryIdEqual;
+		$groupId = $filter->userIdEqual;
+		return $this->deleteCategoryUserSubscriber($categoryId, $groupId);
+	}
+	
+	protected function deleteCategoryUserSubscriber($categoryId, $groupId)
+	{
+		$categoryUserFilter = new KalturaCategoryUserFilter();
+		$categoryUserFilter->categoryIdEqual = $categoryId;
+		
+		$pager = new KalturaFilterPager();
+		$this->pager->pageIndex = 0;
+		$this->pager->pageSize = 100;
+		
+		$categoryUsersList = array();
+		do
+		{
+			$categoryUsersPage = KBatchBase::$kClient->categoryUser->listAction($categoryUserFilter, $pager);
+			$this->pager->pageIndex += 1;
```

Comment:
lets not store all users in memory, instead just run for each one the logic from lines 45+ and gather the groups and subscribers